src/components/LogView/index.js

```diff
@@ -366,7 +369,47 @@ class LogView extends React.Component<Props, State> {
     }
   }
 
+  componentDidMount() {
+    if (this.props.wrap) {
+      this.props.toggleWrap();
+    }
+  }
+
   componentDidUpdate(prevProps: Props, prevState: State) {
+    const currBookmarksSet = new Set(
+      this.props.bookmarks.map(({ lineNumber }) => lineNumber)
+    );
+    const lastLineNo = this.state.lines.length - 1;
+    // handle initial bookmark scroll
+    if (
+      // Don't scroll unless lines are rendered and bookmarks exist
+      this.state.lines.length > 0 &&
+      prevState.lines.length === 0 &&
+      this.props.bookmarks.length &&
+      // Don't scroll when the bookmarks look like [0, last line]
+      !(
+        currBookmarksSet.size === 2 &&
+        currBookmarksSet.has(0) &&
+        currBookmarksSet.has(lastLineNo)
+      ) &&
+      // Don't scroll when we already tried to scroll before lol
+      !this.state.hasScrolledToFirstBookmark
+    ) {
+      const sortedBookmarks = [...this.props.bookmarks].sort(
+        (a, b) => a.lineNumber - b.lineNumber
+      );
+      for (let i = 0; i < sortedBookmarks.length; i++) {
+        const { lineNumber } = sortedBookmarks[i];
+        // don't bother scrolling to first line bc it's visible on render
+        if (lineNumber !== 0) {
+          this.setState({ hasScrolledToFirstBookmark: true }, () => {
+            this.props.scrollToLine(lineNumber);
+          });
+          break;
+        }
+      }
```

Comment:
I'm a little cautious seeing a `setState` in a for loop since it is an anti pattern. Even though it likely doesn't get called more then once with the break i think it would be better to be explicit with its behavior. Something like this.
```
const lineNumber = sortedBookmarks.find(sortedBookmark.linenumber !== 0);

this.setState({hasScrolledToFirstBookmark: true}, () => {
   this.props.scrollToLine(lineNumber);
})
```

Reply:
that's slick 💯